domain/Errors.go

```diff
@@ -9,7 +9,18 @@ var ErrNotAuthorized = errors.New("NotAuthorized")
 var ErrNotAuthenticated = errors.New("NotAuthenticated")
 
 // ErrInvalidRequest is returned when some part of the request is incompatible with another part.
-var ErrInvalidRequest = errors.New("InvalidRequest")
+type ErrInvalidRequest struct {
```

Comment:
Just a note: this really makes me think if we have the right place for this errors class. Nothing that needs to be sorted out right now, but would love to have a short discussion :)